crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -83,55 +123,123 @@ impl Plugin for VisibilityPlugin {
-                .after(TransformSystem::TransformPropagate),
-        )
-        .add_system_to_stage(
-            CoreStage::PostUpdate,
-            update_frusta::<PerspectiveProjection>
-                .label(UpdatePerspectiveFrusta)
-                .after(TransformSystem::TransformPropagate),
-        )
-        .add_system_to_stage(
-            CoreStage::PostUpdate,
-            update_frusta::<Projection>
-                .label(UpdateProjectionFrusta)
-                .after(TransformSystem::TransformPropagate),
-        )
-        .add_system_to_stage(
-            CoreStage::PostUpdate,
-            check_visibility
-                .label(CheckVisibility)
-                .after(CalculateBounds)
-                .after(UpdateOrthographicFrusta)
-                .after(UpdatePerspectiveFrusta)
-                .after(UpdateProjectionFrusta)
-                .after(TransformSystem::TransformPropagate),
-        );
+        app.init_resource::<EntityMeshRelationships>()
+            .add_system_to_stage(
+                CoreStage::PostUpdate,
+                calculate_bounds.label(CalculateBounds),
+            )
+            .add_system_to_stage(CoreStage::PostUpdate, update_bounds.label(UpdateBounds))
+            .add_system_to_stage(
+                CoreStage::PostUpdate,
+                update_frusta::<OrthographicProjection>
+                    .label(UpdateOrthographicFrusta)
+                    .after(TransformSystem::TransformPropagate),
+            )
+            .add_system_to_stage(
+                CoreStage::PostUpdate,
+                update_frusta::<PerspectiveProjection>
+                    .label(UpdatePerspectiveFrusta)
+                    .after(TransformSystem::TransformPropagate),
+            )
+            .add_system_to_stage(
+                CoreStage::PostUpdate,
+                update_frusta::<Projection>
+                    .label(UpdateProjectionFrusta)
+                    .after(TransformSystem::TransformPropagate),
+            )
+            .add_system_to_stage(
+                CoreStage::PostUpdate,
+                check_visibility
+                    .label(CheckVisibility)
+                    .after(CalculateBounds)
+                    .after(UpdateOrthographicFrusta)
+                    .after(UpdatePerspectiveFrusta)
+                    .after(UpdateProjectionFrusta)
+                    .after(TransformSystem::TransformPropagate),
+            );
     }
 }
 
+/// Calculates [`Aabb`]s for [`Entities`](Entity) with [`Mesh`]es. To opt out of bound calculation
+/// for an `Entity`, give it the [`NoAabbCompute`] component.
 pub fn calculate_bounds(
     mut commands: Commands,
     meshes: Res<Assets<Mesh>>,
-    without_aabb: Query<(Entity, &Handle<Mesh>), (Without<Aabb>, Without<NoFrustumCulling>)>,
+    without_aabb: Query<
+        (Entity, &Handle<Mesh>),
+        (
+            Without<Aabb>,
+            Without<NoFrustumCulling>,
+            Without<NoAabbCompute>,
```

Comment:
I don't think this should be here. It makes sense to compute it at least once, I think it's just recomputing the boundsthat should be avoided with this tag.

Reply:
I think it's overkill yes, as you said nobody has complained about having something that can't be calculated. A lot of people have had issues with it not updating automatically though.

The reason I mentioned that marker originally is for situations where you want frustum culling and you know you will update a mesh frequently but the overall size of the mesh won't change. This is technically an optimization that might not be needed though, but I think it makes sense to only have `NoAabbRecompute` (I'd suggest calling it `NoAabbUpdate`, but that doesn't really matter). 

If people don't want any Aabb, then they probably don't want FrustumCulling so just having the `Without<NoFrustumCulling>` should be enough.

Reply:
Renamed to `NoAabbUpdate` and removed it from `calculate_bounds` in 7ac4fec6135b9a4d3ffa83b004fb9b3e889653ba.